subsys/bluetooth/controller/ll_sw/nordic/hal/nrf5/radio/radio_nrf5_dppi_resources.h

```diff
@@ -109,11 +109,11 @@
  * separate PPI is used because packet end is marked by PHYEND event while last
  * bit or CRC is marked by END event.
  */
-#if defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER)
+#if defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) && !defined(CONFIG_BT_CTLR_DF)
```

Comment:
I wonder if `defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) && !defined(CONFIG_BT_CTLR_DF)` or `!defined(CONFIG_BT_CTLR_SW_SWITCH_SINGLE_TIMER) && defined(CONFIG_BT_CTLR_DF)` could be put into a new promptless Kconfig option or otherwise stored as a `#define` or something with a meaningful name?

Reply:
Sure, will give a try.

Reply:
Each conditional compilation logic is a different combination (catering to nrf52-ppi or nrf53/54-DPPI or single timer use), using a Kconfig or #define needs significant refactoring of the implementation. As the implementation has incremental been updated over the years to support changes in SoC h/w, it is now a spaghetti.

Reply:
Yeah, it's pretty hard to follow that many conditionals (some of which are even in other conditionals). 

I can approve for now, but I think some effort needs to be done for this, as it significantly reduces maintainability.